src/create-job-form.tsx

```diff
@@ -252,6 +254,9 @@ export function CreateJobForm(props: CreateJobFormProps): JSX.Element {
           formInput={formInput}
           fields={formFields}
         />
+        {props.customEnvironment &&
+          <props.customEnvironment state={state} setState={setState} />
+        }
```

Comment:
It doesn't look like `props.customEnvironment` will ever be nullish, since our default plugin still returns a component. I'd say we define the type correctly in `CreateJobFormProps` as non-optional.